src/interventions/Interventions.jl

```diff
@@ -1,176 +1,140 @@
-Base.@kwdef struct Intervention{N,P,P2} <: EcoModel
-    # Intervention Parameters
-    # Integer values have a +1 offset to allow for discrete value mapping
-    # (see `set()` and `map_to_discrete()` methods)
+Base.@kwdef struct Intervention{DU,T,DT} <: EcoModel
+    # Intervention Factors
     # Bounds are defined as floats to maintain type stability
-    guided::N = Param(
+    guided::DU = Factor(
         0;
-        ptype="categorical",
-        bounds=(-1.0, length(decision.mcda_methods()) + 1.0),
-        default_bounds=(-1.0, length(decision.mcda_methods()) + 1.0),
-        dists="unif",
-        criteria_keywords=(""),
+        ptype="ordered categorical",
+        dist=DiscreteUniform,
+        dist_params=(-1.0, Float64(length(decision.mcda_methods()))),
         name="Guided",
         description="Choice of MCDA approach.",
     )
-    N_seed_TA::N = Param(
+    N_seed_TA::DU = Factor(
         0;
-        ptype="integer",
-        bounds=(0.0, 1000000.0 + 1.0),
-        default_bounds=(0.0, 1000000.0 + 1.0),
-        dists="unif",
-        criteria_keywords=(""),
+        ptype="unordered categorical",
```

Comment:
I would have thought variables like `N_seed_TA` etc. are ordered categorical because they translate to a scale of magnitude (e.g. the higher the category the larger the number of corals), whereas `guided` I would have thought of as unordered because the categories can go in any order and the meaning is not lost.

Reply:
That's a good point - silly oversight on my part. I will correct.